tests/datasource/fluent/test_spark_filesystem_datasource.py

```diff
@@ -1119,64 +1123,126 @@ def directory_asset(
-        self,
-        directory_asset,
-        column_value_partitioner,
-        expected_num_records_directory_asset_no_partitioner_2020_passenger_count_2: int,
+    def test_yearly_batch_definition_workflow(
+        self, directory_asset, yearly_batch_parameters_and_expected_result
     ):
-        post_partitioner_batch_request = directory_asset.build_batch_request(
-            options={"passenger_count": 2},
-            partitioner=column_value_partitioner,
-        )
-        post_partitioner_batch_list = directory_asset.get_batch_list_from_batch_request(
-            post_partitioner_batch_request
+        batch_parameters, expected_result = yearly_batch_parameters_and_expected_result
+        batch_def = directory_asset.add_batch_definition_yearly(
+            name="yearly", column="pickup_datetime"
         )
-        post_partitioner_batch_data = post_partitioner_batch_list[0].data
+        batch = batch_def.get_batch(batch_parameters=batch_parameters)
+        assert batch.validate(gxe.ExpectTableRowCountToEqual(value=expected_result)).success
 
-        assert (
-            post_partitioner_batch_data.dataframe.count()
-            == expected_num_records_directory_asset_no_partitioner_2020_passenger_count_2
-        )
-        assert (
-            post_partitioner_batch_data.dataframe.filter(F.col("passenger_count") != 2).count() == 0
+    @pytest.mark.spark
+    def test_whole_table_batch_definition_workflow(
+        self, directory_asset, whole_directory_batch_parameters_and_expected_result
+    ):
+        batch_parameters, expected_result = whole_directory_batch_parameters_and_expected_result
+        batch_def = directory_asset.add_batch_definition_whole_directory(name="whole directory")
+        batch = batch_def.get_batch(batch_parameters=batch_parameters)
+        assert batch.validate(gxe.ExpectTableRowCountToEqual(value=expected_result)).success
+
+    @pytest.mark.spark
+    @pytest.mark.parametrize(
+        "partitioner,expected_keys",
+        [
+            pytest.param(
+                PartitionerYearAndMonthAndDay(column_name="foo"),
+                ("path", "year", "month", "day"),
+                id="Daily Partitioner",
+            ),
+            pytest.param(
+                PartitionerYearAndMonth(column_name="foo"),
+                (
+                    "path",
+                    "year",
+                    "month",
+                ),
+                id="Monthly Partitioner",
+            ),
+            pytest.param(
+                PartitionerYear(column_name="foo"),
+                (
+                    "path",
+                    "year",
+                ),
+                id="Yearly Partitioner",
+            ),
+            pytest.param(None, ("path",), id="No Partitioner"),
+        ],
+    )
+    def test_get_batch_parameters_keys_with_partitioner(
+        self, directory_asset, partitioner: Partitioner, expected_keys
```

Comment:
I miss these sometimes too, but probably good to get types on the test params.

Reply:
thanks! updated [c89c7c6](https://github.com/great-expectations/great_expectations/pull/9888/commits/c89c7c660b3a23df35a9abfa70fdc0cf4affa177)